src/sentry/integrations/discord/utils/auth.py

```diff
@@ -0,0 +1,12 @@
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PublicKey
+
+
+def verify_signature(public_key_string: str, signature: str, message: str) -> bool:
+    try:
```

Comment:
Discord's docs recommended a different package for this, but I was able to figure out how to do this with our existing cryptography package! 😄